feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go

```diff
@@ -449,58 +454,103 @@ func TestTraffic(t *testing.T) {
+
+	v6FlowEth := v6Flow.Packet().Add().Ethernet()
+	v6FlowEth.Src().SetValue(session.ATETrafficAttrs.MAC)
+
+	v6FlowIp := v6Flow.Packet().Add().Ipv6()
+	v6FlowIp.Src().SetValue(session.ATETrafficAttrs.IPv6)
+	v6FlowIp.Dst().SetValue(targetNetwork.IPv6)
+
+	// v6Flow.Duration().FixedPackets().SetPackets(100)
+	v6Flow.Rate().SetPps(50)
+	v6Flow.Size().SetFixed(128)
+	v6Flow.Metrics().SetEnable(true)
+
+	deadFlow := ts.ATETop.Flows().Add()
+	deadFlow.SetName("deadFlow")
+	deadFlow.TxRx().Device().SetTxNames([]string{srcIpv4.Name()}).SetRxNames([]string{netv4.Name()})
+
+	deadFlowEth := deadFlow.Packet().Add().Ethernet()
+	deadFlowEth.Src().SetValue(session.ATETrafficAttrs.MAC)
+
+	deadFlowIp := deadFlow.Packet().Add().Ipv4()
+	deadFlowIp.Src().SetValue(session.ATETrafficAttrs.IPv4)
+	deadFlowIp.Dst().SetValue(deadNetwork.IPv4)
+
+	// deadFlow.Duration().FixedPackets().SetPackets(100)
+	deadFlow.Rate().SetPps(50)
+	deadFlow.Size().SetFixed(128)
+	deadFlow.Metrics().SetEnable(true)
+
 	t.Logf("Starting protocols on ATE...")
 	ts.PushAndStart(t)
-	defer ts.ATETop.StopProtocols(t)
 	ts.MustAdjacency(t)
-	t.Logf("Configuring traffic from ATE through DUT...")
-	v4Header := ondatra.NewIPv4Header()
-	v4Header.DstAddressRange().WithMin(targetNetwork.IPv4).WithCount(1)
-	v4Flow := ate.Traffic().NewFlow("v4Flow").
-		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
-		WithHeaders(ondatra.NewEthernetHeader(), v4Header)
-	v6Header := ondatra.NewIPv6Header()
-	v6Header.DstAddressRange().WithMin(targetNetwork.IPv6).WithCount(1)
-	v6Flow := ate.Traffic().NewFlow("v6Flow").
-		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
-		WithHeaders(ondatra.NewEthernetHeader(), v6Header)
-	// deadFlow is addressed to a nonexistent network as a consistency check -
-	// all traffic should be blackholed.
-	deadHeader := ondatra.NewIPv4Header()
-	deadHeader.DstAddressRange().WithMin(deadNetwork.IPv4).WithCount(1)
-	deadFlow := ate.Traffic().NewFlow("flow2").
-		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
-		WithHeaders(ondatra.NewEthernetHeader(), deadHeader)
+
+	time.Sleep(time.Second * 30)
 	t.Logf("Running traffic for 30s...")
-	ate.Traffic().Start(t, v4Flow, v6Flow, deadFlow)
+	otg.StartTraffic(t)
 	time.Sleep(time.Second * 30)
-	ate.Traffic().Stop(t)
-	t.Logf("Checking telemetry...")
-	telem := ate.Telemetry()
-	v4Loss := telem.Flow(v4Flow.Name()).LossPct().Get(t)
-	v6Loss := telem.Flow(v6Flow.Name()).LossPct().Get(t)
-	deadLoss := telem.Flow(deadFlow.Name()).LossPct().Get(t)
-	if v4Loss > 1 {
-		t.Errorf("Got %v%% IPv4 packet loss; expected < 1%%", v4Loss)
-	}
-	if v6Loss > 1 {
-		t.Errorf("Got %v%% IPv6 packet loss; expected < 1%%", v6Loss)
-	}
-	if deadLoss != 100 {
-		t.Errorf("Got %v%% invalid packet loss; expected 100%%", deadLoss)
+	otg.StopTraffic(t)
+
+	t.Logf("Verify Flow metrics...")
+	otgutils.LogFlowMetrics(t, otg, ts.ATETop)
+
+	for _, flow := range ts.ATETop.Flows().Items() {
```

Comment:
Would prefer a helper that returns loss percentage and can be called individually for each flow. That makes it explicit which flows we are checking.

Reply:
Remove the 'for' loop and just call it separately for each flow (since there are only 3 I think the readability is worth it to make explicit the check for each flow.)

Reply:
Done